src/com/sap/piper/tools/ToolDescriptor.groovy

```diff
@@ -0,0 +1,139 @@
+        }
+    }
+
+    def getExecutable(script, configuration, log = true) {
+        def executable = getTool(script, configuration, log)
+        if (log) script.echo "Using $name executable '$executable'."
+        return executable
+    }
+
+    def verify(script, configuration) {
+
+        verifyHome(script, configuration)
+        verifyTool(script, configuration)
+        verifyVersion(script, configuration)
+    }
+
+    def verifyHome(script, configuration) {
+
+        def home = getHome(script, configuration)
+        if (home) { 
+            script.echo "Verifying $name home '$home'."
+            FileUtils.validateDirectoryIsNotEmpty(script, home)
+            script.echo "Verification success. $name home '$home' exists."
+        }
+    }
+
+    def verifyTool(script, configuration) {
+
+        def home = getHome(script, configuration, false)
+        def tool = getTool(script, configuration, false)
+        if (home) {
+            script.echo "Verifying $name '$tool'."
+            FileUtils.validateFile(script, tool)
+            script.echo "Verification success. $name '$tool' exists."
+        }
+    }
+
+    def verifyVersion(script, configuration) {
+
+        def executable = getExecutable(script, configuration, false)
+
+        script.echo "Verifying $name version $version or compatible version."
+
+        def toolVersion
+        try {
+          toolVersion = script.sh returnStdout: true, script: "$executable $versionOption"
+        } catch(AbortException e) {
+          throw new AbortException("The verification of $name failed. Please check '$executable'. $e.message.")
+        }
+        def installedVersion = new Version(toolVersion)
+        if (!installedVersion.isCompatibleVersion(new Version(version))) {
+          throw new AbortException("The installed version of $name is ${installedVersion.toString()}. Please install version $version or a compatible version.")
+        }
+        script.echo "Verification success. $name version ${installedVersion.toString()} is installed."
+    }
+
+    def getConfigurationOptions() {
+        def configOptions = "Please, configure $name home. $name home can be set "
+        if (environmentKey) configOptions += "using the environment variable '$environmentKey', or "
+        if (stepConfigurationKey) configOptions += "using the configuration key '$stepConfigurationKey', or "
+        configOptions += "on PATH."
+        return configOptions
+    }
+
+    def isOnPath(script, configuration) {
+
+        def path
+        try {
+          path = script.sh returnStdout: true, script: """#!/bin/bash --login
+                                                          which $executableName"""
+        } catch(AbortException e) {
+          def exitStatus = script.sh returnStatus: true, script: """#!/bin/bash --login
+                                                                    which $executableName"""
+          if (exitStatus == 1) return false
```

Comment:
Ah yes, can be because when it fails with 1 or more the AbortException is thrown, but then I catch it and ask again, because if the return code is 1, it is not on the path, and I just need to return false, but if it is more than one I better through the AbortException, ans log the error.
Does it make sense, or should I simply return false if it fails? I think that like this is more precise, not?